Automattic/jetpack
# At a Glance: Improve Akismet Upgrade and Activate Flow

Fixes #13035

#### Changes proposed in this Pull Request:

Updates these states:
  * free plan, Akismet not installed
  * free plan, Akismet installed but not activated
  * free plan, Akismet installed and activated but not configured
  * paid plan, Akismet not installed
  * paid plan, Aksimet installed but not activated
  * paid plan, Akismet installed and activated but not configured

The intention is that users on a free plan will clearly see the options both to upgrade to a paid Jetpack plan and to add a key for an existing Akismet subscription, and users on a paid plan with Akismet somehow misconfigured will clearly see the path to get it configured again.

This PR adds a new Jetpack endpoint: `/plugins/akismet/activate`. For now I have hard-coded it to only work with Akismet, because a Jetpack endpoint to activate arbitrary plugins would require a larger discussion and isn't relevant to this project. However, /plugins/akismet/activate would be compatible with such an endpoint.

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?
* This not a new feature and neither adds or removes parts of Jetpack.

#### Testing instructions:





* Go to Jetpack's dashboard.
* In each of the following states, observe the Akismet card and verify that "Upgrade" takes you to buy a Jetpack plan if appropriate and "Activate Akismet" takes you to the Akismet settings page, with both installing and/or activating Akismet as necessary:
    * Jetpack on free plan, Akismet not installed.
    * Jetpack on free plan, Akismet installed but not activated.
    * Jetpack on free plan, Akismet installed and activated.
    * Jetpack on personal plan, Akismet not installed.
    * Jetpack on personal plan, Akismet installed but not activated.
    * Jetpack on personal plan, Akismet installed and activated.
    * Other paid plans should behave the same as the personal plan.

#### Before
<img width="1111" alt="image" src="https://user-images.githubusercontent.com/1123119/61494843-8be74200-a974-11e9-9056-b2b5123d5f83.png">

#### After
<img width="1094" alt="image" src="https://user-images.githubusercontent.com/1123119/61494801-6b1eec80-a974-11e9-8a44-e7a5af54b31f.png">

#### Proposed changelog entry for your changes:

* Improved the flow to purchase and activate Akismet from the At-a-Glance page.
